homeassistant/components/fan/dyson.py

```diff
@@ -233,3 +236,11 @@ def speed_list(self: ToggleEntity) -> list:
     def supported_features(self: ToggleEntity) -> int:
         """Flag supported features."""
         return SUPPORT_OSCILLATE | SUPPORT_SET_SPEED
+
+    @property
+    def state_attributes(self: ToggleEntity) -> dict:
+        """Return optional state attributes."""
+        attributes = super().state_attributes
+        attributes[ATTR_IS_NIGHT_MODE] = self.is_night_mode
```

Comment:
Is this code right? self._device.is_night_mode?

Reply:
Yes the code is correct. The property is defined in [line 185](https://github.com/soraxas/home-assistant/blob/2007cfd49eea90c0527e507f458a6d4bf0215cb4/homeassistant/components/fan/dyson.py#L185). Similar for `self.is_auto_mode`.

Reply:
Oh, i see.